### DISCUSSION TITLE
TerrainLayer lots of requests when zooming out/in

### DISCUSSION THREAD

**Discussion by radoslavirha** [visgl/deck.gl#6091] [Q&A]
Hi,

I tried TerrainLayer with tiled terrain (whole country) and I have an issue with lots of requests when I zoom out, slightly move on the map and zoom in. It seems that older requests are not canceled for previous zoom levels and new requests for current zoom are queued on the end. The texture is blurred for a long time and I see hundreds of requests.

I estimate 75% (from the end of browser's network) of the requests are for my current zoom (when i stop scrolling and wait) which is weird. It calls requests with correct zoom, but for much larger area than my viewport is it seems.

I cleared network before zooming out/in
![image](https://user-images.githubusercontent.com/23722745/129033611-38223032-3450-4239-a3d0-468d9a08eab3.png)

it it known issue, or should I set something? I tried it with basic layer definition from docs https://deck.gl/docs/api-reference/geo-layers/terrain-layer. My elevation model is a few hundred meters above 0 in average.

Thanks

**Comment by Pessimistress** [comment]
Make sure you are using the latest release.

**Reply by radoslavirha** [reply]
Thanks for your reply. It's better now, i can see canceled requests, but in some cases it still loads hundreds of unnecessary tiles. Not sure how it works exactly, but my feeling is I do similar operation few times, sometimes it canceled requests sometimes not.

**Reply by radoslavirha** [reply]
Hi @Pessimistress I tried it again with deck.gl 8.5.5 and loaders.gl 3.0.9 and it's still an issue. There is only few canceled requests. Is there any way how can I manually cancel all the requests for previous zoom level? Thanks

**Reply by kylebarron** [reply]
You might also want to make sure you're putting a sensible value of `maxRequests`, as that determines when requests are aborted.

**Reply by radoslavirha** [reply]
I had default value (6) and I'm using chrome. When I set 3-4 it seems to be slightly better. But still I think there should be much more cancelled requests when I zoom +-10 levels out and 10 +-levels in.
